host/sgx/sgxquote.c

```diff
@@ -622,3 +700,220 @@ oe_result_t oe_sgx_get_supported_attester_format_ids(
+
+        *supplemental_data_size = local_data_size;
+        result = OE_OK;
+    }
+    else
+        // Failed to load DCAP QVL library
+        result = OE_PLATFORM_ERROR;
+
+done:
+    return result;
+}
+
+static oe_result_t sgx_qvl_error_to_oe(quote3_error_t err)
+{
+    switch (err)
+    {
+    case SGX_QL_SUCCESS:
+    case SGX_QL_TCB_SW_HARDENING_NEEDED:
+        return OE_OK;
+
+    case SGX_QL_ERROR_INVALID_PARAMETER:
+        return OE_INVALID_PARAMETER;
+
+    case SGX_QL_PCK_CERT_UNSUPPORTED_FORMAT:
+    case SGX_QL_PCK_CERT_CHAIN_ERROR:
+    case SGX_QL_TCBINFO_UNSUPPORTED_FORMAT:
+    case SGX_QL_TCBINFO_CHAIN_ERROR:
+    case SGX_QL_TCBINFO_MISMATCH:
+    case SGX_QL_QEIDENTITY_UNSUPPORTED_FORMAT:
+    case SGX_QL_QEIDENTITY_CHAIN_ERROR:
+    case SGX_QL_TCB_OUT_OF_DATE:
+    case SGX_QL_SGX_ENCLAVE_IDENTITY_OUT_OF_DATE:
+    case SGX_QL_SGX_ENCLAVE_REPORT_ISVSVN_OUT_OF_DATE:
+    case SGX_QL_QE_IDENTITY_OUT_OF_DATE:
+    case SGX_QL_SGX_TCB_INFO_EXPIRED:
+    case SGX_QL_SGX_PCK_CERT_CHAIN_EXPIRED:
+    case SGX_QL_SGX_SIGNING_CERT_CHAIN_EXPIRED:
+    case SGX_QL_SGX_ENCLAVE_IDENTITY_EXPIRED:
+        return OE_INVALID_ENDORSEMENT;
+
+    case SGX_QL_QUOTE_FORMAT_UNSUPPORTED:
+    case SGX_QL_QE_REPORT_INVALID_SIGNATURE:
+    case SGX_QL_QE_REPORT_UNSUPPORTED_FORMAT:
+    case SGX_QL_QUOTE_CERTIFICATION_DATA_UNSUPPORTED:
+        return OE_QUOTE_VERIFICATION_ERROR;
+
+    case SGX_QL_SGX_CRL_EXPIRED:
+        return OE_VERIFY_CRL_EXPIRED;
+
+    case SGX_QL_PCK_REVOKED:
+    case SGX_QL_TCB_REVOKED:
+        return OE_VERIFY_REVOKED;
+
+    case SGX_QL_TCB_CONFIGURATION_NEEDED:
+    case SGX_QL_TCB_OUT_OF_DATE_CONFIGURATION_NEEDED:
+    case SGX_QL_TCB_CONFIGURATION_AND_SW_HARDENING_NEEDED:
+        return OE_TCB_LEVEL_INVALID;
+
+    default:
+        return OE_UNEXPECTED;
+
+    }
+}
+
+oe_result_t oe_sgx_verify_quote(
+    const oe_uuid_t* format_id,
+    const void* opt_params,
+    size_t opt_params_size,
+    const uint8_t* p_quote,
+    uint32_t quote_size,
+    time_t expiration_check_date,
+    uint32_t* p_collateral_expiration_status,
+    uint32_t* p_quote_verification_result,
+    void* p_qve_report_info,
+    uint32_t qve_report_info_size,
+    void* p_supplemental_data,
+    uint32_t supplemental_data_size,
+    uint32_t collateral_version,
+    const void* p_tcb_info,
+    uint32_t tcb_info_size,
+    const void* p_tcb_info_issuer_chain,
+    uint32_t tcb_info_issuer_chain_size,
+    const void* p_pck_crl,
+    uint32_t pck_crl_size,
+    const void* p_root_ca_crl,
+    uint32_t root_ca_crl_size,
+    const void* p_pck_crl_issuer_chain,
+    uint32_t pck_crl_issuer_chain_size,
+    const void* p_qe_identity,
+    uint32_t qe_identity_size,
+    const void* p_qe_identity_issuer_chain,
+    uint32_t qe_identity_issuer_chain_size)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    quote3_error_t err = SGX_QL_ERROR_UNEXPECTED;
+
+    //TODO add check for format id
```

Comment:
Add github issue # to the comment?  I assume this will be done in step 2?

Reply:
It's a placeholder, I add `format-id` and `opt_param` just want to keep forward compatibility. 
Right now, only ECDSA-p256 is supported. I will update comments.